mono/metadata/boehm-gc.c

```diff
@@ -1545,8 +1556,9 @@ mono_gc_register_finalizer_callbacks (MonoGCFinalizerCallbacks *callbacks)
 		g_error ("Invalid finalizer callback version. Expected %d but got %d\n", MONO_GC_FINALIZER_EXTENSION_VERSION, callbacks->version);
 
 	fin_callbacks = *callbacks;
-
+#if !HAVE_BDWGC_GC
 	GC_set_await_finalize_proc ((void (*) (GC_PTR))fin_notifier);
```

Comment:
The function mono_gc_register_finalizer_callbacks doesn't appear to be used anywhere in mono...just exposed via mono.def